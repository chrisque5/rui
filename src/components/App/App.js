import moment from 'moment-timezone';
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Route, HashRouter, Switch } from 'react-router-dom';
import { ConfigProvider } from 'antd';
import { LicenseManager } from 'ag-grid-enterprise';
import { initUserData } from '../../actions/userActions';
import { loadLicenseKey } from '../../actions/systemSettingsActions';
import { getLocale } from '../../utils/localeHelper';
import { getUserFullName, getLicenseKey } from '../../utils/selectors';
import { saveCurrentPageAndRedirect, clearWindowName } from '../../utils/windowHelper';

import Home from '../Home/Home';
import Deal from '../Deal/Deal';
import DealEdit from '../DealEdit/DealEdit';
import Admin from '../Admin/Admin';
import NotFound from '../Error/NotFound';
import ErrorBoundary from '../Error/ErrorBoundary';
import BlotterContainer from '../Blotter/BlotterContainer';

const App = () => {
  const dispatch = useDispatch();
  const userName = useSelector(getUserFullName);
  const AG_GRID_LICENSE = useSelector(getLicenseKey);

  useEffect(() => {
    moment.locale(navigator.language || navigator.languages[0] || 'en-GB');
    moment.tz.setDefault('Europe/London');
    dispatch(loadLicenseKey());

    async function fetchUserData() {
      await dispatch(initUserData());
    }
    fetchUserData();
  }, [dispatch]);

  useEffect(() => {
    if (AG_GRID_LICENSE) { LicenseManager.setLicenseKey(AG_GRID_LICENSE); }
  }, [AG_GRID_LICENSE]);

  useEffect(() => {
    // Clearining window name if previously exists
    clearWindowName();
    if (userName !== '' && localStorage.getItem('loggedInUser') !== userName) {
      localStorage.setItem('loggedInUser', userName);
    }
    // It will trigger when local storage changed in another window
    window.onstorage = () => {
      if (!localStorage.getItem('loggedInUser')) {
        window.close();
        // Capturing the current page if window not closed
        saveCurrentPageAndRedirect(userName);
      }
    };
  }, [userName]);

  const routes = [
    { path: '/', exact: true, component: Home },
    { path: '/deal/:dealType', component: Deal },
    { path: '/dealedit/:dealId', component: DealEdit },
    { path: '/admin', component: Admin },
    { path: '/blotter', component: BlotterContainer },
    { path: '*', component: NotFound },
  ];

  return (
    <ConfigProvider locale={getLocale()}>
      <ErrorBoundary>
        <HashRouter>
          <Switch>
            {
              routes.map((route) => {
                const { component, exact, path } = route;
                return (<Route key={path} path={path} exact={exact} component={component} />);
              })
            }
          </Switch>
        </HashRouter>
      </ErrorBoundary>
    </ConfigProvider>
  );
};

export default App;
