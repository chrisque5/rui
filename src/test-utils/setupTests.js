// The below part of the code is to fix the issue
// TypeError: window.matchMedia is not a function
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});
