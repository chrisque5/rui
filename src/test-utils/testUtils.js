/* eslint-disable import/no-extraneous-dependencies, react/prop-types */
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Form } from 'antd';

const middlewares = [thunk];
const renderWithForm = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}><Form>{children}</Form></Provider>;

  return render(ui, { wrapper, ...renderOptions });
};

const renderWithProvider = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}>{children}</Provider>;

  return render(ui, { wrapper, ...renderOptions });
};
// re-export everything
export * from '@testing-library/react';
// override render method
export { renderWithForm, renderWithProvider };
