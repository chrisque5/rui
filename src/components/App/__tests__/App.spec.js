import { renderWithProvider } from 'test-utils/testUtils';
import App from '../App';

const initialState = {};

describe('<App />', () => {
  test('renders snapshot as expected', () => {
    const { container } = renderWithProvider(<App />, { initialState });
    expect(container).toMatchSnapshot();
  });
});
