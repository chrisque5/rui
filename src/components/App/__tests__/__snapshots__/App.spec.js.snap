// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders snapshot as expected 1`] = `
<div>
  <div
    class="dms-trade-capture-navbar ant-menu-dark"
  >
    <ul
      class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-dark navbar-menu"
      data-menu-list="true"
      role="menu"
      tabindex="0"
    >
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-no-link"
        data-menu-id="rc-menu-uuid-test-home"
        role="menuitem"
        style="opacity: 1; order: 0;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <img
            alt="logo"
            height="30"
            src="tpicap.png"
            style="margin-right: 15px;"
            width="30"
          />
          <span>
            Deal Management System
          </span>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-submenu ant-menu-submenu-horizontal user-name"
        role="none"
        style="opacity: 1; order: 1;"
      >
        <div
          aria-controls="rc-menu-uuid-test-submenu-popup"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            <span
              class="submenu-title-wrapper"
              data-testid="ddlUserDropdown"
              id="ddlUserDropdown"
            >
              
              <span
                aria-label="caret-down"
                class="anticon anticon-caret-down"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="caret-down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 1024 1024"
                  width="1em"
                >
                  <path
                    d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        aria-hidden="true"
        class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
        role="none"
        style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
      >
        <div
          aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          data-menu-id="rc-menu-uuid-test-rc-menu-more"
          role="menuitem"
          tabindex="-1"
        >
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
    <div
      aria-hidden="true"
      style="display: none;"
    />
  </div>
  0
</div>
`;
